### ISSUE TITLE
invalid argument '-install_name @rpath/libdemo.1.0.dylib' only allowed with '-dynamiclib'

### ISSUE THREAD

**Issue by mdavidsaver** [epics-base/setuptools_dso#34]
It looks like the logic to replace `-bundle` with `-dynamiclib` has been perturbed, so the GHA jobs for OSX are failing.  Presumably due to a change in setuptools or its bundled distutils some time in the past 2 weeks.

```
  g++ -Wno-unused-result -Wsign-compare -Wunreachable-code -fno-common -dynamic -DNDEBUG -g -fwrapv -O3 -Wall \
 -arch arm64 -arch x86_64 -g -bundle -undefined dynamic_lookup build/temp.macosx-11-universal2-cpython-38/src/foo.o \
build/temp.macosx-11-universal2-cpython-38/src/bar.o -o build/lib.macosx-11-universal2-cpython-38/dsodemo \
/lib/libdemo.1.0.dylib -install_name @rpath/libdemo.1.0.dylib
  clang: error: invalid argument '-install_name @rpath/libdemo.1.0.dylib' only allowed with '-dynamiclib'
  clang: error: invalid argument '-install_name @rpath/libdemo.1.0.dylib' only allowed with '-dynamiclib'
  error: command '/usr/bin/g++' failed with exit code 1
  error: subprocess-exited-with-error
```

https://github.com/epics-base/setuptools_dso/blob/8b28347d172040bb226e30673698171fa75c63e5/src/setuptools_dso/dsocmd.py#L348-L352

**Comment by mdavidsaver** [comment]
pypa/setuptools@0ab156c3e12054503a0dac0ce172e241262881f4 introduces `compiler.linker_so_cxx`.

**Comment by mdavidsaver** [comment]
Fixed with 2.11
